py/server/deephaven/filters.py

```diff
@@ -2,23 +2,90 @@
 #   Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
 #
 """ This module implement various filters that can be used in deephaven table's filter operations."""
+from __future__ import annotations
+
+from typing import List, Union
 
 import jpy
 
 from deephaven import DHError
 from deephaven._wrapper import JObjectWrapper
+from deephaven.jcompat import to_sequence
 
 _JFilter = jpy.get_type("io.deephaven.api.filter.Filter")
 _JRegexFilter = jpy.get_type("io.deephaven.engine.table.impl.select.RegexFilter")
+_JFilterOr = jpy.get_type("io.deephaven.api.filter.FilterOr")
+_JFilterAnd = jpy.get_type("io.deephaven.api.filter.FilterAnd")
+_JFilterNot = jpy.get_type("io.deephaven.api.filter.FilterNot")
 
 
 class Filter(JObjectWrapper):
+    """A Filter object represents a filter that can be used in Table's filtering(where) operations."""
     j_object_type = _JFilter
 
     @property
     def j_object(self) -> jpy.JType:
         return self.j_filter
 
+    def __init__(self, j_filter):
+        self.j_filter = j_filter
+
+    @classmethod
+    def from_(cls, conditions: Union[str, List[str]]) -> Union[Filter, List[Filter]]:
+        """Creates filter(s) from the given condition(s).
+
+        Args:
+            conditions (Union[str, List[str]]): filter condition(s)
+
+        Returns:
+            filter(s)
+
+        Raises:
+            DHError
+        """
+        conditions = to_sequence(conditions)
+        try:
+            filters = [cls(j_filter=j_filter) for j_filter in _JFilter.from_(conditions).toArray()]
+            return filters if len(filters) != 1 else filters[0]
+        except Exception as e:
+            raise DHError(e, "failed to create filters.") from e
+
+
+def or_(filters: List[Filter]) -> Filter:
```

Comment:
Looking at this, and unit tests, it looks like we have:

```python
my_table.where(or_(Filter.from_(["a > 100", "b < 1000", "c < 0"])))
```

I'm wondering if we don't want something like:

```python
my_table.where(or_(["a > 100", "b < 1000", "c < 0"]))
```

And, it might not just be strings, but filters themselves:

```python
f1 = and_(["a > 100", "b <1000"])
f2 = and_(["c > 1", "d < 1"])
f = or_([f1, f2, "z>42"])
my_table.where(f)
```

Reply:
Not sure we want to allow mixing Filter objects with conditional strings. Arguably it might be more confusing than helpful.

Reply:
I'm fine w/ the conclusion as long as we are happy.
```python
f1 = and_(Filter.from_(["a > 100", "b <1000"]))
f2 = and_(Filter.from_(["c > 1", "d < 1"]))
f = or_([f1, f2, Filter.from_("z>42")])
my_table.where(f)
```